storage/client_range_test.go

```diff
@@ -327,22 +327,53 @@ func TestRangeLookupUseReverse(t *testing.T) {
 	defer stopper.Stop()
 
 	// Init test ranges:
-	// ["1","a"), ["a","c"), ["c","e"), ["e","g") and ["g","\xff\xff").
+	// ["","a"), ["a","c"), ["c","e"), ["e","g") and ["g","\xff\xff").
 	splits := []proto.AdminSplitRequest{
 		adminSplitArgs(proto.Key("g"), proto.Key("g"), 1, store.StoreID()),
 		adminSplitArgs(proto.Key("e"), proto.Key("e"), 1, store.StoreID()),
 		adminSplitArgs(proto.Key("c"), proto.Key("c"), 1, store.StoreID()),
 		adminSplitArgs(proto.Key("a"), proto.Key("a"), 1, store.StoreID()),
-		adminSplitArgs(proto.Key("1"), proto.Key("1"), 1, store.StoreID()),
 	}
 
+	txn := &proto.Transaction{ID: uuid.NewUUID4(), Timestamp: store.Clock().Now()}
 	for _, split := range splits {
+		split.Txn = txn
 		_, err := store.ExecuteCmd(context.Background(), &split)
 		if err != nil {
 			t.Fatalf("%q: split unexpected error: %s", split.SplitKey, err)
 		}
 	}
 
+	// Resolve the intents.
+	rArgs := proto.ResolveIntentRangeRequest{
```

Comment:
could you resolve the intents with a `Scan` (over the whole meta key range) and `SucceedsWithin` instead? That should be shorter and less confusing (admittedly, not faster).